aspnet/fundamentals/error-handling.rst

```diff
@@ -1,8 +1,175 @@
+
+The next tab shows the query string parameters, if any:
+
+.. image:: error-handling/_static/developer-exception-page-query.png
+
+In this case, you can see the value of the ``throw`` parameter that was passed to this request. This request didn't have any cookies, but if it did, they would appear on the Cookies tab. You can see the headers that were passed in the last tab:
+
+.. image:: error-handling/_static/developer-exception-page-headers.png
+
+.. _status-code-pages:
+
+Configuring Status Code Pages
+-----------------------------
+
+By default, your app will not provide a rich status code page for HTTP status codes such as 500 (Internal Server Error) or 404 (Not Found). You can configure the ``StatusCodePagesMiddleware`` adding this line to the ``Configure`` method:
+
+.. code-block:: c#
+
+  app.UseStatusCodePages();
+
+By default, this middleware adds very simple, text-only handlers for common status codes. For example, the following is the result of a 404 Not Found status code:
+
+.. image:: error-handling/_static/default-404-status-code.png
+
+The middleware supports several different extension methods. You can pass it a custom lamba expression:
+
+.. code-block:: c#
+
+  app.UseStatusCodePages(context => 
+    context.HttpContext.Response.SendAsync("Handler, status code: " +
+    context.HttpContext.Response.StatusCode, "text/plain"));
+
+Alternately, you can simply pass it a content type and a format string:
+
+.. code-block:: c#
+
+  app.UseStatusCodePages("text/plain", "Response, status code: {0}");
+
+The middleware can handle redirects (with either relative or absolute URL paths), passing the status code as part of the URL:
+
+.. code-block:: c#
+
+  app.UseStatusCodePagesWithRedirects("~/errors/{0}");
+
+In the above case, the client browser will see a ``302 / Found`` status and will redirect to the URL provided.
+
+Alternately, the middleware can re-execute the request from a new path format string:
+
+.. code-block:: c#
+
+  app.UseStatusCodePagesWithReExecute("/errors/{0}");
+
+The ``UseStatusCodePagesWithReExecute`` method will still return the original status code to the browser, but will also execute the handler given at the path specified.
+
+If you need to disable status code pages for certain requests, you can do so using the following code:
+
+.. code-block:: c#
+
+  var statusCodePagesFeature = context.Features.Get<IStatusCodePagesFeature>();
+  if (statusCodePagesFeature != null)
+  {
+    statusCodePagesFeature.Enabled = false;
+  }
+
+Limitations of Exception Handling During Client-Server Interaction
+------------------------------------------------------------------
+
+Web apps have certain limitations to their exception handling capabilities because of the nature of disconnected HTTP requests and responses. Keep these in mind as you design your app's exception handling behavior.
+
+#. Once the headers for a response have been sent, you cannot change the response's status code.
```

Comment:
nor can any exception pages or handlers run. The response must be completed or the connection aborted.